sympy/functions/combinatorial/tests/test_comb_factorials.py

```diff
@@ -351,61 +379,119 @@ def test_factorial2_rewrite():
     assert factorial2(2*n + 1).rewrite(gamma) == \
         sqrt(2)*2**(n + S.Half)*gamma(n + Rational(3, 2))/sqrt(pi)
 
+@XFAIL
+def test_binomial_fail1():
+    i, j = symbols('i, j', integer=True)
+    # doesn't hold for (i, j) == (0, 0)
+    assert binomial(i, j).equals(
+        binomial(i - 1, j - 1) + binomial(i - 1, j)) is not True
+
+
+@XFAIL
+def test_binomial_fail2():
+    i, j = symbols('i, j', integer=True)
+    # gives None but should be True
+    assert binomial(2*j, j).equals(
+        (-4)**j*binomial(-S.Half, j)) is True
+
+
+@slow
+@XFAIL
+def test_binomial_fail3():
+    i, j = symbols('i j', integer=True, zero=False)
+    # gives None but should be Tru
+    assert binomial(i, j).equals(
+        binomial(i - 1, j - 1) + binomial(i - 1, j))
+
 
 def test_binomial():
     x = Symbol('x')
-    n = Symbol('n', integer=True)
-    nz = Symbol('nz', integer=True, nonzero=True)
-    k = Symbol('k', integer=True)
+
+    n, k = symbols('n k', integer=True)
     kp = Symbol('kp', integer=True, positive=True)
     kn = Symbol('kn', integer=True, negative=True)
+    kne = Symbol('kn', integer=True, negative=True, even=True)
+    nt, kt = symbols('nt kt', integer=False)
+    nr = symbols('nr', rational=False, real=True)
+    a, b = symbols('a b', integer=True, nonnegative=True)
     u = Symbol('u', negative=True)
     v = Symbol('v', nonnegative=True)
     p = Symbol('p', positive=True)
     z = Symbol('z', zero=True)
-    nt = Symbol('nt', integer=False)
-    kt = Symbol('kt', integer=False)
-    a = Symbol('a', integer=True, nonnegative=True)
-    b = Symbol('b', integer=True, nonnegative=True)
-
-    assert binomial(0, 0) == 1
-    assert binomial(1, 1) == 1
-    assert binomial(10, 10) == 1
-    assert binomial(n, z) == 1
-    assert binomial(1, 2) == 0
-    assert binomial(-1, 2) == 1
-    assert binomial(1, -1) == 0
-    assert binomial(-1, 1) == -1
-    assert binomial(-1, -1) == 0
-    assert binomial(S.Half, S.Half) == 1
-    assert binomial(-10, 1) == -10
-    assert binomial(-10, 7) == -11440
-    assert binomial(n, -1) == 0 # holds for all integers (negative, zero, positive)
-    assert binomial(kp, -1) == 0
-    assert binomial(nz, 0) == 1
-    assert expand_func(binomial(n, 1)) == n
-    assert expand_func(binomial(n, 2)) == n*(n - 1)/2
-    assert expand_func(binomial(n, n - 2)) == n*(n - 1)/2
-    assert expand_func(binomial(n, n - 1)) == n
-    assert binomial(n, 3).func == binomial
+    nz = Symbol('z', zero=False)
```

Comment:
```suggestion
    nz = Symbol('nz', zero=False)
```
this seems to have snuck in

Reply:
It's best (but not necessary) for names to match the variable used.

Reply:
there was already a z in use for zero=True